cmd/sample-cosi-driver/main.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The Kubernetes Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig := <-sigs
+		klog.InfoS("Signal received", "type", sig)
+		cancel()
+
+		<-time.After(30 * time.Second)
```

Comment:
What purpose does 30s wait before exiting serve?

Reply:
it allows for the cancelled context to gracefully shutdown